### PR TITLE
Update DNS doc

### PR DESCRIPTION
**Pull request by thockin**

### CODE REVIEW

File: docs/dns.md [hunk 1]
```diff
@@ -1,51 +1,31 @@
...
+As of kubernetes 0.8, DNS is offered as a cluster add-on.  If enabled, a DNS
+Pod and Service will be scheduled on the cluster, and the kubelets will be
+configured to tell individual containers to use the DNS Service's IP.
+
+Every Service defined in the cluster (including the DNS server itself) will be
+assigned a DNS name.  By default, a client Pod's DNS search list will
+include the Pod's own namespace and the cluster's default domain.  This is best
+illustrated by example.
+
+Assume a Service named `foo` in namespace `bar`.  A Pod running in namespace
```
Comment (erictune) [inline docs/dns.md]: Be completely explicit here and say "Kubernetes namespace", to disambiguate from DNS namespace.  They happen to match, but the reader doesn't know that yet.

File: docs/dns.md [hunk 2]
```diff
@@ -1,51 +1,31 @@
...
-and create a service called `blog` in Kubernetes.  Then external DNS requests to `blog.mydomain.com` will
-automatically resolve to the service proxy and be forwarded to the pods running Wordpress.
-
-Full documentation of the SkyDNS server is in the [SkyDNS repository](https://github.com/skynetservices/skydns)
-and abbreviated information is available by typing `skydns --help`.
+
+The DNS pod that runs holds 3 containers - skydns, etcd (which skydns uses),
+and a kubernetes-to-skydns bridge called kube2sky.  The kube2sky process
+watches the kubernetes master for changes in Services, and then writes the
+information to etcd, which skydns reads.
```
Comment (erictune) [inline docs/dns.md]: Clarify that the etcd pod used by skydns is not linked to the etcd used by the master, or any other etcd in the cluster.

File: docs/dns.md [hunk 3]
```diff
@@ -1,51 +1,31 @@
...
+
+## Issues
+
+While DNS resolution works from nodes directly (outside of a container), nodes
```
Comment (erictune) [inline docs/dns.md]: Are you saying that `nslookup foo.bar` doesn't work from an uncontainerized node command line, but `nslookup foo.bar $SKYDNS_IP` does?  If so maybe give that example, or phrase as "while the skydns server is reachable by uncontainerized processes on the node, it is not in the search path."

File: docs/dns.md [hunk 4]
```diff
@@ -1,51 +1,31 @@
...
+# DNS Integration with Kubernetes
+
+As of kubernetes 0.8, DNS is offered as a cluster add-on.  If enabled, a DNS
+Pod and Service will be scheduled on the cluster, and the kubelets will be
+configured to tell individual containers to use the DNS Service's IP.
+
+Every Service defined in the cluster (including the DNS server itself) will be
+assigned a DNS name.  By default, a client Pod's DNS search list will
+include the Pod's own namespace and the cluster's default domain.  This is best
+illustrated by example.
```
Comment (dchen1107) [inline docs/dns.md]: Link to the example here?

Comment (dchen1107) [inline docs/dns.md]: Ahh, you mean example below. That is fine.

File: docs/dns.md (hunk 1 above)
Comment (thockin) [inline docs/dns.md]: Done

File: docs/dns.md (hunk 2 above)
Comment (thockin) [inline docs/dns.md]: Done

File: docs/dns.md (hunk 3 above)
Comment (thockin) [inline docs/dns.md]: Reworded, PTAL

**Comment by thockin** [comment]
All comments addressed (I think)

**Comment by thockin** [comment]
ping

**Comment by thockin** [comment]
Self-merging, since I am on-call.

**Pull request merged**